File: packaging/npm/bin/tide.js
#!/usr/bin/env node
/*
TideMark
========

File: packaging/npm/bin/tide.js
Description: npm command entrypoint for forwarding to the native TideMark `tide` binary.

Responsibility:
- Delegate CLI invocation to resolved native executable.

Architectural Position:
- Node command shim for package-manager installations.

Author: Silan.Hu
Email: silan.hu@u.nus.edu
Copyright (c) 2026-2027 easynet. All rights reserved.
*/

const { runBinary } = require("../lib/runner");

runBinary("tide", process.argv.slice(2)).catch((error) => {
  console.error(`error: ${error.message}`);
  process.exitCode = 1;
});

File: src/bin/tide.rs
//! TideMark
//! ========
//!
//! File: src/bin/tide.rs
//! Description: Primary TideMark CLI binary bootstrap.
//!
//! Responsibility:
//! - Parse command-line input and dispatch execution to the application runner.
//!
//! Architectural Position:
//! - Executable entrypoint for direct TideMark command usage.
//!
//! Author: Silan.Hu
//! Email: silan.hu@u.nus.edu
//! Copyright (c) 2026-2027 easynet. All rights reserved.

use std::process::ExitCode;

use clap::Parser;
use tidemark::{app, interface::cli::Cli};

fn main() -> ExitCode {
    let cli = Cli::parse();
    match app::run(cli) {
        Ok(()) => ExitCode::SUCCESS,
        Err(err) => {
            eprintln!("error: {err}");
            err.exit_code()
        }
    }
}
